### PR TITLE
Camunda process selection + registration backend core

### PR DESCRIPTION
**Pull request by sergei-maertens**
Related to #326 

**Changes**

* Added API endpoint for staff members to read the available process instances
* Added custom, dynamic registration backend options form in UI to select a process definition
    * Optionally, a specific version (i.e. - not the "latest") can be selected

### CODE REVIEW

**Comment by sergei-maertens** [comment]
@SilviaAmAm - I would appreciate a frontend-code review!

File: src/openforms/js/components/admin/form_design/registrations/camunda.js [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    return (
+        <Wrapper>
+            <CustomFieldTemplate
+                id="camundaOptions.processDefinition"
+                label={intl.formatMessage({
+                    defaultMessage: 'Process',
+                    description: 'Camunda \'process definition\' label'
+                })}
+                rawErrors={null} errors={null} // TODO
+                required
```
Comment (SilviaAmAm) [inline src/openforms/js/components/admin/form_design/registrations/camunda.js:100]: The `processDefinition` is missing the help text (`rawDescription` field in the `processDefinitionVersion`)

**Review by SilviaAmAm — CHANGES_REQUESTED** [review]

File: src/openforms/js/components/admin/form_design/registrations/camunda.js [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+
+const useLoadProcessDefinitions = () => {
+    let processDefinitions;
+    const { loading, value, error } = useAsync(
+        async () => {
+            const response = await get(PROCESS_DEFINITIONS_ENDPOINT);
+            if (!response.ok) throw new Error(`Response status: ${response.status}`);
+            return response.data;
+        }, [PROCESS_DEFINITIONS_ENDPOINT]
```
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/openforms/js/components/admin/form_design/registrations/camunda.js:24]: it's not, but this ensures that the hook only runs once. I should use `[]` instead though, that has the same impact.

File: src/openforms/js/components/admin/form_design/RegistrationFields.js [hunk 3]
```diff
@@ -7,6 +7,47 @@ import FormRow from '../forms/FormRow';
 import Fieldset from '../forms/Fieldset';
 import Select from '../forms/Select';
 import FormRjsfWrapper from '../RJSFWrapper';
+import {BACKEND_OPTIONS_FORMS} from './registrations';
+
+
+const BackendType = PropTypes.shape({
```
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/openforms/js/components/admin/form_design/RegistrationFields.js:13]: good suggestion, won't pollute this PR with that cleanup though. It also makes things a bit worse - it helps to keep the type defs in sync, but seeing the shape of data entering the component now requires you to open another file.

File: src/openforms/js/lang/nl.json [hunk 4]
```diff
@@ -49,6 +49,11 @@
     "description": "Registration backend options label",
     "originalDefault": "Registration backend options"
   },
+  "2uIYst": {
+    "defaultMessage": "The process definition for which to start a process instance.",
```
Comment (joeribekker) [inline src/openforms/js/lang/nl.json:53]: Make my life easier by not having to search for untranslated texts...

```suggestion
    "defaultMessage": "De procesdefinitie waarvoor een proces instantie gestart moet worden.",
```

File: src/openforms/js/lang/nl.json [hunk 5]
```diff
@@ -324,6 +329,11 @@
     "description": "Form slug field help text",
     "originalDefault": "Slug of the form, used in URLs"
   },
+  "Hfdfv2": {
+    "defaultMessage": "Which version of the process definition to start. The latest version is used if not specified.",
```
Comment (joeribekker) [inline src/openforms/js/lang/nl.json:333]:
```suggestion
    "defaultMessage": "Welke versie van de procesdefinitie moet gebruikt worden. Indien niet opgegeven dan wordt altijd de laatste versie gebruikt.",
```

File: src/openforms/js/lang/nl.json [hunk 6]
```diff
@@ -429,6 +439,11 @@
     "description": "Admin save and add another submit button",
     "originalDefault": "Save and add another"
   },
+  "PaEo2j": {
+    "defaultMessage": "Process",
```
Comment (joeribekker) [inline src/openforms/js/lang/nl.json:443]:
```suggestion
    "defaultMessage": "Proces",
```

File: src/openforms/js/lang/nl.json [hunk 7]
```diff
@@ -779,6 +794,11 @@
     "description": "Logic trigger number of months",
     "originalDefault": "months"
   },
+  "n4Y2ex": {
+    "defaultMessage": "Version",
```
Comment (joeribekker) [inline src/openforms/js/lang/nl.json:798]:
```suggestion
    "defaultMessage": "Versie",
```

**Review by joeribekker — APPROVED** [review]
Looks good but please commit my translations :)

**Pull request merged**